File: Data/data_plot.py
# -*- coding: utf-8 -*-

import sys
import cmath
import math

import numpy as np
import matplotlib.pyplot as plt
import matplotlib.gridspec as gridspec
import matplotlib.ticker as tick


#グラフの書式設定 -------------------------------------------------------------
plt.rcParams["font.family"] = "Arial"                #全体のフォントを設定
plt.rcParams['mathtext.fontset'] = 'cm'              #数式のフォントを設定
plt.rcParams["font.size"] = 12                       #フォントの大きさ
plt.rcParams['axes.labelsize'] = 16                  #ラベルのフォントの大きさ
plt.rcParams["xtick.direction"] = "in"               #x軸の目盛線が内向き('in')か外向き('out')か双方向か('inout')
plt.rcParams["ytick.direction"] = "in"               #y軸の目盛線が内向き('in')か外向き('out')か双方向か('inout')
plt.rcParams["xtick.minor.visible"] = False          #x軸補助目盛りがあり(True)かなし(False)か
plt.rcParams["ytick.minor.visible"] = False          #y軸補助目盛りがあり(True)かなし(False)か
plt.rcParams["xtick.major.width"] = 1.0              #x軸主目盛り線の線幅
plt.rcParams["ytick.major.width"] = 1.0              #y軸主目盛り線の線幅
plt.rcParams["xtick.minor.width"] = 1.0              #x軸補助目盛り線の線幅
plt.rcParams["ytick.minor.width"] = 1.0              #y軸補助目盛り線の線幅
plt.rcParams["xtick.major.size"] = 6                 #x軸主目盛り線の長さ
plt.rcParams["ytick.major.size"] = 6                 #y軸主目盛り線の長さ
plt.rcParams["xtick.minor.size"] = 5                 #x軸補助目盛り線の長さ
plt.rcParams["ytick.minor.size"] = 5                 #y軸補助目盛り線の長さ
plt.rcParams["axes.linewidth"] = 1.0                 #囲みの太さ

LINE_WIDTH = 0.5
T_MIN = 10
T_MAX = 40
# -----------------------------------------------------------------------------

file1 = "9.30/20200930_4.csv"

# header = open(file,'r').readline().strip().split(',')
data = np.loadtxt(file1, delimiter=",", skiprows=1)

time = data[:, 0]
arduino_time = data[:, 1]
body_z = data[:, 2]
R_temp = data[:, 3]
R_add_cur = data[:, 4]
R_tgt_cur = data[:, 5]
R_cur = data[:, 6]
R_vel = data[:, 7]
R_pos = data[:, 8]

L_temp = data[:, 9]
L_add_cur = data[:, 10]
L_tgt_cur = data[:, 11]
L_cur = data[:, 12]
L_vel = data[:, 13]
L_pos = data[:, 14]


# 16bit Encoder
LSB = 65536
RGR = 6

R_torq = R_cur/2000*12.5*3.3
R_vel = R_vel*RGR
R_pos = R_pos/600
R_tgt_torq = R_tgt_cur/2000*12.5*3.3

L_torq = L_cur/2000*12.5*3.3
L_vel = L_vel*RGR
L_pos = L_pos/600
L_tgt_torq = L_tgt_cur/2000*12.5*3.3

plt.figure(figsize=(7,5))
plt.plot(time, R_tgt_torq, label= "R Target Torque", lw = LINE_WIDTH)
plt.plot(time, R_torq, label= "R Current Torque", lw = LINE_WIDTH)
plt.plot(time, L_tgt_torq, label= "L Target Torque", lw = LINE_WIDTH)
plt.plot(time, L_torq, label= "L Current Torque", lw = LINE_WIDTH)
plt.xlim(T_MIN, T_MAX)
plt.legend()

plt.figure(figsize=(7,5))
plt.plot(time, R_vel, label= "R Current Velocity", lw = LINE_WIDTH)
plt.plot(time, L_vel, label= "L Current Velocity", lw = LINE_WIDTH)
plt.xlim(T_MIN, T_MAX)
plt.legend()

plt.figure(figsize=(7,5))
plt.plot(time, R_pos, label= "R Current Poistion", lw = LINE_WIDTH)
plt.plot(time, L_pos, label= "L Current Poistion", lw = LINE_WIDTH)
plt.xlim(T_MIN, T_MAX)
plt.ylim(-100, 100)
plt.legend()

plt.figure(figsize=(7,5))
plt.plot(time, body_z, label= "body", lw = LINE_WIDTH)
plt.xlim(T_MIN, T_MAX)
plt.legend()

plt.show()

File: Data/data_plot.m
%% initial settings
clear
close all

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
set(0,'defaultAxesFontSize',16);
set(0,'defaultAxesFontName', 'times');
set(0,'defaultTextFontSize',16);
set(0,'defaultTextFontName', 'times');
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%% csv読み込み
file = importdata('10.14\output2.csv', ',', 1);

time = file.data(:, 1);
arduino_time = file.data(:, 2);
sync = file.data(:, 3);
R_temp = file.data(:, 4);
R_add_cur = file.data(:, 5);
R_tgt_cur = file.data(:, 6);
R_cur = file.data(:, 7);
R_vel = file.data(:, 8);
R_pos = file.data(:, 9);

L_temp = file.data(:, 10);
L_add_cur = file.data(:, 11);
L_tgt_cur = file.data(:, 12);
L_cur = file.data(:, 13);
L_vel = file.data(:, 14);
L_pos = file.data(:, 15);

% 16bit Encoder
LSB = 65536;
RGR = 6;

R_torq = R_cur/2000*12.5*3.3;
R_vel = R_vel*RGR;
R_tgt_torq = R_tgt_cur/2000*12.5*3.3;

L_torq = L_cur/2000*12.5*3.3;
L_vel = L_vel*RGR;
L_tgt_torq = L_tgt_cur/2000*12.5*3.3;

figure
plot(time, R_tgt_torq, 'DisplayName', "R_tgt_torq")
hold on
plot(time, R_torq, 'DisplayName', "R_torq")
plot(time, R_pos, 'DisplayName', "R_pos")
legend

figure
plot(time, L_tgt_torq, 'DisplayName', "L_tgt_torq")
hold on
plot(time, L_torq, 'DisplayName', "L_torq")
plot(time, L_pos, 'DisplayName', "L_pos")
legend

% filename = ['outputFig'];        
% saveas(gcf, filename, 'png')